bson/src/main/org/bson/codecs/configuration/CycleDetectingCodecRegistry.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.configuration;
+
+import org.bson.codecs.Codec;
+
+/**
+ * A marker interface for {@code CodecRegistry} implementations that are able to detect cycles.
+ *
+ * @since 3.12
+ */
+public interface CycleDetectingCodecRegistry extends CodecRegistry {
```

Comment:
Necessary to make this a public API so that ChildCodecRegistry can be made generic to the concrete type of the CodecRegistry.

Reply:
After moving the new codec registry into the bson module this could remain internal